### PR TITLE
DCM-60: Add a button to test the DHIS Connection status

### PR DESCRIPTION
**Pull request by Piumal1999**
## The issue I worked on
See [https://issues.openmrs.org/browse/DCM-60](https://issues.openmrs.org/browse/DCM-60) 

## Description of what I changed:
Added a button to test the DHIS Connection status. When the test connection button is pressed, it will display check whether the connection is successful or failure.

## Preview:
https://vimeo.com/581871639

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/dhisconnector/web/controller/DHISConnectorController.java [hunk 1]
```diff
@@ -144,6 +144,24 @@ public void saveConfig(ModelMap model, @RequestParam(value = "url", required = t
...
+	@RequestMapping(value = "/module/dhisconnector/configureServer", params = "testConfig", method = RequestMethod.POST)
+	public void testConfig(ModelMap model, WebRequest req) throws ParseException {
+
+		AdministrationService as = Context.getAdministrationService();
+		String url = as.getGlobalPropertyObject(GLOBAL_PROPERTY_URL).getPropertyValue();
+		String user = as.getGlobalPropertyObject(GLOBAL_PROPERTY_USER).getPropertyValue();
+		String pass = as.getGlobalPropertyObject(GLOBAL_PROPERTY_PASS).getPropertyValue();
+
+		if (Context.getService(DHISConnectorService.class).testDHISServerDetails(url, user, pass)) {
+			req.setAttribute(WebConstants.OPENMRS_MSG_ATTR, "Successfully Connected to DHIS2", WebRequest.SCOPE_SESSION);
```
Comment (HerbertYiga) [inline omod/src/main/java/org/openmrs/module/dhisconnector/web/controller/DHISConnectorController.java:156]: thanks @Piumal1999  why not add (Successfully Connected to DHIS2) under messages.properties?

File: omod/src/main/java/org/openmrs/module/dhisconnector/web/controller/DHISConnectorController.java [hunk 2]
```diff
@@ -144,6 +144,24 @@ public void saveConfig(ModelMap model, @RequestParam(value = "url", required = t
...
+		} else {
+			req.setAttribute(WebConstants.OPENMRS_ERROR_ATTR, "Connection Failed", WebRequest.SCOPE_SESSION);
```
Comment (HerbertYiga) [inline omod/src/main/java/org/openmrs/module/dhisconnector/web/controller/DHISConnectorController.java:158]: same here !!

File: omod/src/main/java/org/openmrs/module/dhisconnector/web/controller/DHISConnectorController.java (hunk 1 above)
Comment (Piumal1999) [reply omod/src/main/java/org/openmrs/module/dhisconnector/web/controller/DHISConnectorController.java]: Ah yes. I'll do it like that. Thanks for the suggestion.
Fixed with 8e157ab

File: omod/src/main/java/org/openmrs/module/dhisconnector/web/controller/DHISConnectorController.java (hunk 2 above)
Comment (Piumal1999) [reply omod/src/main/java/org/openmrs/module/dhisconnector/web/controller/DHISConnectorController.java]: Thanks for the suggestion.
Fixed with 8e157ab

**Comment by akshika47** [comment]
Thank you @Piumal1999

**Pull request merged**